src/document-submit.js

```diff
@@ -0,0 +1,111 @@
+/**
```

Comment:
The goal is to avoid any circumstances where this handler gets installed twice. But please respond to @cramforce 's note on whether or not this is a service. Generally, though, I believe we've been using services for this kind of features.